databricks/koalas/indexes.py

```diff
@@ -2086,6 +2086,50 @@ def item(self):
         """
         return self.to_series().item()
 
+    def insert(self, loc: int, item):
+        """
+        Make new Index inserting new item at location.
+
+        Follows Python list.append semantics for negative values.
+
+        Parameters
+        ----------
+        loc : int
+        item : object
+
+        Returns
+        -------
+        new_index : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([1, 2, 3, 4, 5])
+        >>> kidx.insert(3, 100)
+        Int64Index([1, 2, 3, 100, 4, 5], dtype='int64')
+
+        For negative values
+
+        >>> kidx = ks.Index([1, 2, 3, 4, 5])
+        >>> kidx.insert(-3, 100)
+        Int64Index([1, 2, 100, 3, 4, 5], dtype='int64')
+        """
+        length = len(self)
```

Comment:
If the given `loc` is positive, I think we don't have to calculate the total length here:

```python
>>> ks.Series([1,2,3])[:5]
0    1
1    2
2    3
dtype: int64
>>> ks.Series([1,2,3])[5:]
Series([], dtype: int64)
```

Reply:
Oh, thanks!